provider/pkg/gen/typegen.go

```diff
@@ -463,339 +463,360 @@ func makeSchemaType(prop map[string]any, canonicalGroups map[string]string) stri
 // --------------------------------------------------------------------------
 
 func createGroups(definitionsJSON map[string]any, allowHyphens bool) []GroupConfig {
```

Comment:
This function was refactored by splitting it up into multiple sub-functions to make it easier to unit test each sub-action. This made it easier to track down what changes were required so that schema generation in p-k was ultimately not modified.